Modules/Loadable/SubjectHierarchy/Widgets/qMRMLSubjectHierarchyComboBox.cxx

```diff
@@ -299,21 +308,60 @@ void qMRMLSubjectHierarchyComboBox::showPopup()
   QPoint below = mapToGlobal(listRect.bottomLeft());
   QPoint above = mapToGlobal(listRect.topLeft());
 
-  // Custom size
+  // Custom Height
+  int popupHeight = 0;
   int displayedItemCount = d->TreeView->displayedItemCount();
-  const int numberOfRows = qMin(displayedItemCount, d->MaximumNumberOfShownItems);
-  QSize itemSize = QSize(
-    d->TreeView->sizeHintForColumn(d->TreeView->model()->nameColumn()), d->TreeView->sizeHintForRow(0) );
-  listRect.setHeight( numberOfRows * itemSize.height() );
+  if (displayedItemCount == 0)
+    {
+    // If there is no items, find what message to show instead
+    vtkMRMLSubjectHierarchyNode* shNode = d->TreeView->subjectHierarchyNode();
+    vtkIdType rootItem = d->TreeView->rootItem();
+    std::vector<vtkIdType> childItemIDs;
+    shNode->GetItemChildren(rootItem, childItemIDs, false);
+    if (childItemIDs.empty())
+      {
+      if (rootItem!= shNode->GetSceneItemID())
+        {
+        d->NoItemLabel->setText("No items in branch");
+        }
+      else
+        {
+        d->NoItemLabel->setText("No items in scene");
+        }
+      }
+    else
+      {
+      d->NoItemLabel->setText("No items accepted by current filters");
```

Comment:
Do you think it's safe to make a claim about filters without actually checking them? Maybe though...

Reply:
I thought about that. I first wrote some code to check if filters were applied, but that implied changes in model and tree view classes, and when using it here I realized there was no `else` left : if there are no items, and the root item has children, then they have to had been filtered out. Makes it a little simpler and lighter.